tools/publish-automation/publish.go

```diff
@@ -0,0 +1,269 @@
+			return false
+		}
+
+		// myJson.close
+
+		err = json.Unmarshal(myJson, &pck)
+		if err != nil {
+			fmt.Println("Error unmarshalling JSON:", err)
+			// myJson.
+			return false
+		}
+		if pck.Version == version {
+			fmt.Println("JSON at ./tools/fleetctl-npm/package.json: " + pck.Version + " is coherent with current version: " + version)
+			fmt.Println("Hit ENTER for next step")
+			fmt.Scanln()
+			// myJson.close
+			return true
+		}
+
+		// myJson.close
+		fmt.Println("JSON at ./tools/fleetctl-npm/package.json shows: " + pck.Version + " which is not coherent with current version: " + version)
+		fmt.Println("Please fix ./tools/fleetctl-npm/package.json and hit ENTER.")
+		fmt.Scanln()
+	}
+}
+
+func openReleasesPage(version string) bool {
+	fmt.Println("\n===========================================================")
+	fmt.Println("Step 3: Publish the release in Fleet Releases page.")
+	fmt.Println("Hitting ENTER will take you to the proper editing page (of current release)")
+	fmt.Println("Once published, go back here and hit ENTER again")
+	fmt.Scanln()
+	fmt.Println("Taking you to the edit page ... ")
+
+	url := "https://github.com/fleetdm/fleet/releases/edit/fleet-" + version
+	err := openURL(url)
+	if err != nil {
+		fmt.Println("Error opening URL:", err)
+		return false
+	}
+
+	fmt.Println("Hit ENTER for next step")
+	fmt.Scanln()
+	return true
+}
+
+func openURL(url string) error {
+	var cmd *exec.Cmd
+
+	switch runtime.GOOS {
+	case "windows":
+		cmd = exec.Command("cmd", "/c", "start", url)
+	case "darwin":
+		cmd = exec.Command("open", url)
+	default:
+		cmd = exec.Command("xdg-open", url)
+	}
+
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func npmPublish() bool {
+	fmt.Println("\n===========================================================")
+	fmt.Println("Step 4: NPM PUBLISH")
+	fmt.Println("Hitting ENTER will take you to another terminal to fleet/tools/fleetctl-npm/")
+	fmt.Println("Once there type \"npm publish\". You will need an auth code for that.")
+	fmt.Scanln()
+	fmt.Println("Taking you to the other terminal ... ")
+
+	relativePath := "./tools/fleetctl-npm"
+	err := openTerminalAtPath(relativePath)
+	if err != nil {
+		fmt.Println("Error opening Terminal:", err)
+		return false
+	}
+
+	fmt.Println("Hit ENTER for next step")
+	fmt.Scanln()
+	return true
+}
+
+func openTerminalAtPath(path string) error {
+	var cmd *exec.Cmd
+
+	switch runtime.GOOS {
+	case "darwin":
+		cmd = exec.Command("open", "-a", "Terminal", path)
+	default:
+		return fmt.Errorf("unsupported operating system")
+	}
+
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkDocker(version string) bool {
+	fmt.Println("\n===========================================================")
+	fmt.Println("Step 5: Check hub.docker for existing version")
+	fmt.Println("Hitting ENTER will take you to the proper hub.docker page.")
+	fmt.Println("Verify that the version exists")
```

Comment:
The problem wasn't that the image didn't exist. You pushed the fleetctl image instead of the fleet server image.

When the image doesn't exist it fails more gracefullly

Reply:
Correct.
This will take the user to the proper image (And will give the correct image next step) so will prevent such mistakes.

Reply:
again if we are running this script at the time this is ready to deploy we can achieve it in scripting w/ `gh`

Reply:
My goal with scripts is that they are more than just a replica of documentation. Since maintaining both is more effort.

Reply:
https://github.com/fleetdm/fleet/pull/17870 added into publish step of patch_release script